src/commands/lambda/flare.ts

```diff
@@ -1,14 +1,249 @@
+      errorFound = true
+    }
+
+    if (errorFound) {
+      return 1
+    }
+
+    // Get AWS credentials
+    this.context.stdout.write('\n🔑 Getting AWS credentials...\n')
+    try {
+      this.credentials = await getAWSCredentials()
+    } catch (err) {
+      this.context.stderr.write(commonRenderer.renderError(err))
+
+      return 1
+    }
+    if (this.credentials === undefined) {
+      this.context.stdout.write(commonRenderer.renderNoAWSCredentialsFound())
+      try {
+        await requestAWSCredentials()
+      } catch (err) {
+        this.context.stderr.write(commonRenderer.renderError(err))
+
+        return 1
+      }
+    }
+
+    // Get Lambda function configuration
+    this.context.stdout.write('\n🔍 Getting Lambda function configuration...\n')
+    const lambdaClientConfig: LambdaClientConfig = {
+      region,
+      credentials: this.credentials,
+    }
+    const lambdaClient = new LambdaClient(lambdaClientConfig)
+    let config
+    try {
+      config = await getLambdaFunctionConfig(lambdaClient, this.functionName ?? '')
+    } catch (err) {
+      this.context.stderr.write(
+        commonRenderer.renderError(`Unable to get Lambda function configuration: ${err.message}`)
+      )
+
+      return 1
+    }
+
+    // Print config
+    const configStrColored = util.inspect(config, false, undefined, true)
+    this.context.stdout.write(`\n${configStrColored}\n`)
+    if (this.isDryRun) {
+      this.context.stdout.write('\n🚫 The configuration was not sent as it was executed in dry run mode.\n')
+
+      return 0
+    }
+
+    // Send data to Datadog
+    const folderPath = path.join(process.cwd(), FLARE_OUTPUT_DIRECTORY)
+    const filePath = path.join(folderPath, FUNCTION_CONFIG_FILE_NAME)
+    const zipPath = path.join(folderPath, ZIP_FILE_NAME)
+    const configStrUncolored = JSON.stringify(config, undefined, 2)
+    try {
+      await writeFile(folderPath, filePath, configStrUncolored, this.context)
+      await zipContents(filePath, zipPath)
+      this.context.stdout.write('\n🚀 Sending to Datadog Support...\n')
+      await sendToDatadog(zipPath, this.caseId!, this.email!, this.apiKey!, this.context)
+      deleteFolder(folderPath, this.context)
+    } catch (err) {
+      this.context.stderr.write(commonRenderer.renderError(err.message))
+
+      return 1
+    }
+
+    return 0
+  }
+}
+
+/**
+ * Write the function config to a file
+ * @param folderPath
+ * @param filePath
+ * @param data
+ * @param context used for logging
+ * @throws Error if the file cannot be written
+ */
+export const writeFile = async (folderPath: string, filePath: string, data: string, context: BaseContext) => {
+  if (fs.existsSync(folderPath)) {
+    deleteFolder(folderPath, context)
+  }
+
+  try {
+    fs.mkdirSync(folderPath)
+    fs.writeFileSync(filePath, data)
+  } catch (err) {
+    throw Error(`Unable to save function config: ${err.message}`)
```

Comment:
This is doing two things but we are printing an specific error. What if the error was creating the folder?

```suggestion
    throw Error(`Unable to create function configuration file: ${err.message}`)
```

Reply:
Or if we want to be consistent with the other errors `Unable to create flare file: ...`